docs/src/main/asciidoc/includes/tracing/common-callbacks.adoc

```diff
@@ -0,0 +1,161 @@
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+///////////////////////////////////////////////////////////////////////////////
+
+ifndef::rootdir[:rootdir: {docdir}/..]
+include::{rootdir}/attributes.adoc
+
+// tag::defs[]
+:tracing-javadoc: {tracing-javadoc-base-url}/io/helidon/tracing
+:tracing-api-javadoc: {tracing-javadoc}
+:listener-type: SpanListener
+// end::defs[]
+
+
+// tag::detailed[]
+// tag::intro[]
+== Responding to Span Lifecycle Events [[Tracing-callbacks]]
+
+Applications and libraries can register listeners to be notified at several moments during the lifecycle of every span:
+
+* Before a new span starts
+* After a new span has started
+* After a span ends
+* After a span is activated (creating a new scope)
+* After a scope is closed
+
+// end::intro[]
+
+The next sections explain how you can write and add a listener and what it can do. See the link:{tracing-api-javadoc}/{listener-type}.html[`{listener-type}`] Javadoc for more information.
+
+=== Understanding What Listeners Do
+A listener cannot affect the lifecycle of a span or scope it is notified about, but it can add tags and events and update the baggage associated with a span.
+Often a listener does additional work that does not change the span or scope such as logging a message.
+
+When Helidon invokes the listener's methods it passes proxies for the parameter types. These proxies limit the access the listener has to the span builder, span, or scope, as summarized in the following table. If a listener method tries to invoke an forbidden operation, the proxies throw an `UnsupportedOperationException` and Helidon then logs a `WARNING` message describing the invalid operation invocation.
+
+.Summary of Permitted Operations on Proxies Passed to Listeners
+|====
+| Tracing type | Changes allowed
+
+| link:{tracing-javadoc}/Span.Builder.html[`Span.Builder`]  | Add tags
+
+| link:{tracing-javadoc}/Span.html[`Span`] |  Retrieve and update baggage, add events, add tags
+
+| link:{tracing-javadoc}/Scope.html[`Scope`]  | none
+
+|====
+
+The following tables list specifically what operations the proxies permit.
+
+
+.link:{tracing-javadoc}/Span.Builder.html[`io.helidon.tracing.Span.Builder`] Operations
+|====
+| Method | Purpose | OK?
+
+| `build()` | Starts the span. | -
+| `end` methods | Ends the span. | -
+| `get()` | Starts the span. | -
+| `kind(Kind)` | Sets the "kind" of span (server, client, internal, etc.) | -
+| `parent(SpanContext)` | Sets the parent of the span to be created from the builder. | -
+| `start()` | Starts the span. | -
+| `start(Instant)` | Starts the span. | -
+| `tag` methods| Add a tag to the builder before the span is built. | &check;
+| `unwrap(Class)` | Cast the builder to the specified implementation type. † | &check;
+|====
+
+† Helidon returns the unwrapped object, not a proxy to it.
```

Comment:
Nit: my English is failing me but shouldn't this be "proxy for"?

Reply:
I strongly doubt that your English is failing you. Fixed.